feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -349,29 +350,46 @@ class BillingWrapper(
-        @BillingClient.BillingResponseCode responseCode: Int,
+        isSuccessful: Boolean,
         purchasesByHashedToken: Map<String, PurchaseWrapper>
-    ) : QueryPurchasesResult(responseCode, purchasesByHashedToken) {
+    ) : QueryPurchasesResult(isSuccessful, purchasesByHashedToken)
 
-        override fun isSuccessful(): Boolean = responseCode == BillingClient.BillingResponseCode.OK
-    }
+    private fun Purchase.PurchasesResult.isSuccessful() = responseCode == BillingClient.BillingResponseCode.OK
 
-    override fun queryPurchases(@SkuType skuType: String): QueryPurchasesResult? {
-        return billingClient?.let { billingClient ->
-            log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE.format(skuType))
-            val result = billingClient.queryPurchases(skuType)
+    override fun queryPurchases(
+        appUserID: String,
+        completion: (QueryPurchasesResult) -> Unit
+    ) {
+        billingClient?.let { billingClient ->
+            log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE)
 
-            val purchasesList = result.purchasesList ?: emptyList<Purchase>()
+            val queryActiveSubscriptionsResult = billingClient.queryPurchases(SkuType.SUBS)
+            val queryUnconsumedInAppsRequest = billingClient.queryPurchases(SkuType.INAPP)
 
-            GoogleQueryPurchasesResult(
-                result.responseCode,
-                purchasesList.map { purchase ->
-                    val hash = purchase.purchaseToken.sha1()
-                    log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE_WITH_HASH.format(skuType, hash))
-                    hash to GooglePurchaseWrapper(purchase, skuType.toProductType(), presentedOfferingIdentifier = null)
-                }.toMap()
-            )
-        }
+            val isSuccessful =
+                queryActiveSubscriptionsResult.isSuccessful() && queryUnconsumedInAppsRequest.isSuccessful()
+
+            val activeSubscriptionsList = queryActiveSubscriptionsResult.purchasesList ?: emptyList<Purchase>()
+            val mapOfActiveSubscriptions = activeSubscriptionsList.toMapOfGooglePurchaseWrapper(SkuType.SUBS)
+
+            val unconsumedInAppsList = queryUnconsumedInAppsRequest.purchasesList ?: emptyList<Purchase>()
+            val mapOfUnconsumedInApps = unconsumedInAppsList.toMapOfGooglePurchaseWrapper(SkuType.INAPP)
+
+            completion(GoogleQueryPurchasesResult(
+                isSuccessful,
+                mapOfActiveSubscriptions + mapOfUnconsumedInApps
+            ))
+        } ?: completion(GoogleQueryPurchasesResult(isSuccessful = false, emptyMap()))
+    }
+
+    private fun List<Purchase>.toMapOfGooglePurchaseWrapper(
+        @SkuType skuType: String
+    ): Map<String, GooglePurchaseWrapper> {
+        return this.map { purchase ->
+            val hash = purchase.purchaseToken.sha1()
+            log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE_WITH_HASH.format(skuType, hash))
```

Comment:
I realize that this code was already here before this PR, but I feel like this log is an unexpected side-effect if you're calling `toMapOfGooglePurchaseWrapper`, maybe it could be moved outside of the method?

Reply:
Totally agree. The code is full of these unexpected logs.